### PR TITLE
Update README and remove the backwards compatibility document

### PR DESCRIPTION
**Pull request by xmudrii**
**What this PR does / why we need it**:

Update README and remove the backwards compatibility document.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/assign @kron4eg

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,187 +4,86 @@
...
-  vSphere and Azure
-
-## Installing KubeOne
-
-### Downloading a binary from GitHub Releases
-
-The fastest way to get KubeOne:
-```bash
-curl -sfL get.kubeone.io | sh
-```
```
Comment (kron4eg) [inline README.md:57]: Probably we need to save this

Comment (xmudrii) [reply README.md:57]: Do we really need it if we pointed to the documentation a bit down?

Comment (kron4eg) [reply README.md:57]: Yeah, it's for the marketing reasons

Comment (xmudrii) [reply README.md]: @kron4eg Fixed, PTAL.

**Pull request merged**